src/jailer/src/main.rs

```diff
@@ -274,18 +277,42 @@ pub fn to_cstring<T: AsRef<Path>>(path: T) -> Result<CString> {
     CString::new(path_str).map_err(Error::CStringParsing)
 }
 
-fn main() {
+#[derive(Debug)]
+struct JailerExitError {
+    exit_code: ExitCode,
+    error_msg: String,
+}
+
+impl Termination for JailerExitError {
+    fn report(self) -> ExitCode {
+        eprintln!("{}", self.error_msg);
+        self.exit_code
+    }
+}
+
+impl JailerExitError {
+    fn new(exit_code: ExitCode, error_msg: String) -> Self {
+        JailerExitError {
+            exit_code,
+            error_msg,
+        }
+    }
+}
+
+fn main() -> core::result::Result<(), JailerExitError> {
```

Comment:
This should return a custom error type (same comment as on `run_with_api` https://github.com/firecracker-microvm/firecracker/pull/3575/commits/f83b7c8cd3fdb3b320ee1fe1f80769c4c498ee56#r1163827837).